Feature/top ten countries by indicator

weberjavi opened the pull request:
This PR adds top ten countries chart to Countries Context by indicator section on Agriculture profile.
Pivotal task:
https://www.pivotaltracker.com/story/show/163084689
Designs:
https://projects.invisionapp.com/d/main#/console/16383011/339735320/preview

![kapture 2019-02-06 at 17 31 17](https://user-images.githubusercontent.com/6906348/52356820-39a10d00-2a35-11e9-8a5c-10e25ce05f00.gif)

aabdaab commented:
The countries chart is looking good 👍 
I found a bug in the map tooltip though:
![image](https://user-images.githubusercontent.com/6136899/52358590-2fc8db00-2a30-11e9-85fb-c135f93ebc86.png)
When there's no data it should say `No data` (in some cases it's working correctly).

aabdaab commented:
Also, on smaller screens I'm experiencing this bug with tooltip staying when scrolling away from the map, I'm not sure if it's just me, could you check on your side?
![tooltipbug](https://user-images.githubusercontent.com/6136899/52360170-4de40a80-2a33-11e9-8972-10f468292fb0.gif)

aabdaab commented on app/javascript/app/components/sectors-agriculture/countries-context/context-by-indicator/context-by-indicator-component.jsx:
```diff
@@ -49,60 +50,105 @@ const ContextByIndicatorComponent = ({
+          </div>
+        </div>
+        <div className={styles.visualizationsContainer}>
+          <div>
```
I think we don't need that extra div here

weberjavi replied:
It is used as container for Map and Legend, I've added a placeholder `layout-container` class to be explicit that they are a `column` layout element.

aabdaab commented on app/javascript/app/components/sectors-agriculture/countries-context/context-by-indicator/context-by-indicator-component.jsx:
```diff
@@ -49,60 +50,105 @@ const ContextByIndicatorComponent = ({
+            <Map
+              paths={paths}
+              tooltipId="cc-map-tooltip"
+              onCountryClick={() => {}}
```
maybe if there's no explicit action to do in this case we could pass `undefined` instead of an empty function?

weberjavi replied:
⚡️

kfilip94 commented:
I don't know why but for e.g. `Male employment` in 2002: Chad (82.3%) has shorter bar then Rwanda where value is 82%: 
![screenshot from 2019-02-06 17-42-31](https://user-images.githubusercontent.com/15097138/52361752-df08b080-2a36-11e9-9c92-31231adea964.png)

weberjavi commented:
> The countries chart is looking good 👍
> I found a bug in the map tooltip though:
> ![image](https://user-images.githubusercontent.com/6136899/52358590-2fc8db00-2a30-11e9-85fb-c135f93ebc86.png)
> When there's no data it should say `No data` (in some cases it's working correctly).

So this happens in countries that where not sovereign in the year selected (It happens on the USSR, Yugoslavia, Ethiopia and Sudan), they only have available data or `no data` from their independence year onwards.
How should we handle this?
Any proposal?

aabdaab commented:
Javi, so the issue here is that for some of the `iso`s you get no data for a given year, right? Then maybe we could iterate through the paths, after updating them with what's in the API, and filling the ones that are empty with the label `No data`

simaob commented:
A couple of other small tweaks, if you don't mind me barging in: 

- sort indicators' dropdown alphabetically;
- sort years' dropdown from most recent to latest;

Tks!

weberjavi commented:
☝️ addressed comments.
There's one thing left behind though. The map tooltip not being hide after scroll on mobile.
After trying to make it work through `react-tooltip` event handlers (`event`, `eventOff` and `globalEventOff`) and failing I've just adjusted the `delayHide` property, but as scrolling should fire the `hide` function it just makes the tooltip to remain there for two more second after the `scroll` event is over.
Further investigation would be needed to address this bug.
Do you agree that we handle it in another PR?

simaob commented:
Yes, I think it's OK to handle that in another PR.

simaob commented:
I've added it.
